lib/internal/async_local_storage/native.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const {
+  ReflectApply,
+} = primordials;
+
+const AsyncContextFrame = require('internal/async_context_frame');
+const { AsyncResource } = require('async_hooks');
+
+class AsyncLocalStorage {
+  static bind(fn) {
+    return AsyncResource.bind(fn);
+  }
+
+  static snapshot() {
+    return AsyncLocalStorage.bind((cb, ...args) => cb(...args));
+  }
+
+  disable() {
+    AsyncContextFrame.disable(this);
+  }
+
+  enterWith(data) {
+    const frame = new AsyncContextFrame(this, data);
+    AsyncContextFrame.set(frame);
+  }
+
+  run(data, fn, ...args) {
+    const prior = AsyncContextFrame.current();
+    this.enterWith(data);
+    try {
+      return ReflectApply(fn, null, args);
+    } finally {
+      AsyncContextFrame.set(prior);
+    }
+  }
+
+  exit(fn, ...args) {
```

Comment:
This variant of `exit` seems quite different to `exit` in other implementation as it no longer disables the store.

Reply:
The behaviour is functionally equivalent as a disabled store simply returns `undefined` when calling `getStore()`. There's not really an internally equivalent way to do the disable with CPED as all you can do is set and get a value slot, so the way to achieve functional equivalence from the external API perspective is to store `undefined`.

Reply:
actually this is mostly the same topic as that one below so resolving this thread.